tensorflow/python/ops/array_grad.py

```diff
@@ -641,7 +641,7 @@ def _GatherV2Grad(op, grad):
   # For axis 0 gathers, build an appropriately shaped IndexedSlices.
   if axis_static == 0:
     if context.executing_eagerly():
-      params_tail_shape = params_shape.cpu()[1:]
+      params_tail_shape = array_ops.identity(params_shape)[1:]
```

Comment:
Doesn't this need a `with ops.device("cpu")`: as well, otherwise this is not forcing it onto the CPU

Reply:
@alextp: Is this CPU placement necessary? I looked at the history of this code and couldn't quite figure out why the CPU placement is needed.